DebugGUI/src/DebugGUI.cxx

```diff
@@ -51,9 +54,80 @@ void* initGUI(const char* name, void(*error_callback)(int, char const*descriptio
   io.Fonts->AddFontFromMemoryTTF((void*)s_iconsFontAwesomeTtf, sizeof(s_iconsFontAwesomeTtf), 12.0f, &icons_config, icons_ranges);
 
   ImPlot::CreateContext();
+
   return window;
 }
 
+// returns a string containing drawing data in JSON format
+// the returned JSON is composed of a list of draw commands
+std::string getFrameJSON(ImDrawData *draw_data) {
+  std::ostringstream json_data;
```

Comment:
This means you will create the json_data at each frame. I would pass an ostream& as an argument.